src/backend/opengl/CMakeLists.txt

```diff
@@ -1,156 +1,130 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.0)
-
-FIND_PACKAGE(OpenGL REQUIRED)
-
-INCLUDE_DIRECTORIES(${OPENGL_INCLUDE_DIR})
-
-OPTION(USE_SYSTEM_GLBINDING "Use system glbinding" OFF)
-IF(USE_SYSTEM_GLBINDING)
-    # Point glbinding_DIR (case sensitive) to the location of glbinding-config.cmake
-    # This file is generally at CMAKE_INSTALL_PREFIX/glbinding-config.cmake of
-    # the glbinding project
-    FIND_PACKAGE(glbinding REQUIRED)
-ELSE(USE_SYSTEM_GLBINDING)
-    INCLUDE(build_glbinding)
-ENDIF(USE_SYSTEM_GLBINDING)
-
-IF(WIN32)
-    ADD_DEFINITIONS(-DFGDLL)
-ENDIF(WIN32)
-
-INCLUDE_DIRECTORIES(
-    ${CMAKE_INCLUDE_PATH}
-    ${WTK_INCLUDE_DIRS}
-    ${CMAKE_CURRENT_SOURCE_DIR}
-    "${PROJECT_SOURCE_DIR}/src/api/c"
-    "${PROJECT_SOURCE_DIR}/src/api/cpp"
-    "${PROJECT_SOURCE_DIR}/src/backend"
-    )
+find_package(OpenGL REQUIRED)
 
-FILE(GLOB api_headers
-    "${PROJECT_SOURCE_DIR}/include/*.h"
-    "${PROJECT_SOURCE_DIR}/include/fg/*.h"
-    )
+set(BackendTarget "forge")
 
-FILE(GLOB c_api_headers
-    "${PROJECT_SOURCE_DIR}/src/api/c/*.hpp"
-    )
-FILE(GLOB c_api_sources
-    "${PROJECT_SOURCE_DIR}/src/api/c/*.cpp"
-    )
-FILE(GLOB cpp_api_headers
-    "${PROJECT_SOURCE_DIR}/src/api/cpp/*.hpp"
-    )
-FILE(GLOB cpp_api_sources
-    "${PROJECT_SOURCE_DIR}/src/api/cpp/*.cpp"
-    )
+add_library(${BackendTarget} SHARED "")
 
-FILE(GLOB bknd_headers
-    "${PROJECT_SOURCE_DIR}/src/backend/*.hpp"
-    )
+set_target_properties(${BackendTarget} PROPERTIES FOLDER Forge)
+set_target_properties(${BackendTarget} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(${BackendTarget} PROPERTIES CXX_STANDARD 11)
+set_target_properties(${BackendTarget} PROPERTIES LINKER_LANGUAGE CXX)
 
-FILE(GLOB bknd_sources
-    "${PROJECT_SOURCE_DIR}/src/backend/*.cpp"
+target_compile_definitions(${BackendTarget}
+    PRIVATE
+        WIN32_MEAN_AND_LEAN=$<PLATFORM_ID:Windows>
+        OS_WIN=$<PLATFORM_ID:Windows>
```

Comment:
it would be cleaner if you defined only the platform you are compiling for. I think its safe to use add_definition for this type of thing.